code/modules/shuttle/supply.dm

```diff
@@ -108,131 +115,543 @@
+			GLOB.station_money_database.charge_account(account, -credit_changes[account], "Supply Shuttle Fine", "Central Command Supply Master", allow_overdraft = TRUE, supress_log = FALSE)
 
 	SSeconomy.centcom_message += "[msg]<hr>"
-	if(credits_to_deposit > 0)
-		GLOB.station_money_database.credit_account(SSeconomy.cargo_account, credits_to_deposit, "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
-		if(research_credits)
-			GLOB.station_money_database.credit_account(GLOB.station_money_database.get_account_by_department(DEPARTMENT_SCIENCE), research_credits, "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
-		if(service_credits)
-			GLOB.station_money_database.credit_account(GLOB.station_money_database.get_account_by_department(DEPARTMENT_SERVICE), service_credits, "Supply Shuttle Exports Payment", "Central Command Supply Master", supress_log = FALSE)
+	manifest = new
+
+
+// Convenience object that registers itself with the supply shuttle and provides
+// methods for you to override.
+/datum/economy/simple_seller
+
+/datum/economy/simple_seller/proc/register(obj/docking_port/mobile/supply/S)
+	RegisterSignal(S, COMSIG_CARGO_BEGIN_SCAN,			PROC_REF(begin_scan))
+	RegisterSignal(S, COMSIG_CARGO_CHECK_SELL,			PROC_REF(check_sell))
+	RegisterSignal(S, COMSIG_CARGO_BEGIN_SELL,			PROC_REF(begin_sell))
+	RegisterSignal(S, COMSIG_CARGO_DO_PRIORITY_SELL,	PROC_REF(sell_priority))
+	RegisterSignal(S, COMSIG_CARGO_DO_SELL,				PROC_REF(sell_normal))
+	RegisterSignal(S, COMSIG_CARGO_SEND_ERROR,			PROC_REF(sell_wrong))
+	RegisterSignal(S, COMSIG_CARGO_END_SELL,			PROC_REF(end_sell))
+
+/datum/economy/simple_seller/proc/begin_scan(obj/docking_port/mobile/supply/S)
+	SIGNAL_HANDLER
+	return
+
+/datum/economy/simple_seller/proc/check_sell(obj/docking_port/mobile/supply/S, atom/movable/AM)
+	SIGNAL_HANDLER
+	return NONE
+
+/datum/economy/simple_seller/proc/begin_sell(obj/docking_port/mobile/supply/S)
+	SIGNAL_HANDLER
+	return
+
+/datum/economy/simple_seller/proc/sell_priority(obj/docking_port/mobile/supply/S, atom/movable/AM, datum/economy/cargo_shuttle_manifest/manifest)
+	SIGNAL_HANDLER
+	return check_sell(S, AM) & COMSIG_CARGO_SELL_PRIORITY
+
+/datum/economy/simple_seller/proc/sell_normal(obj/docking_port/mobile/supply/S, atom/movable/AM, datum/economy/cargo_shuttle_manifest/manifest)
+	SIGNAL_HANDLER
+	return check_sell(S, AM) & COMSIG_CARGO_SELL_NORMAL
+
+/datum/economy/simple_seller/proc/sell_wrong(obj/docking_port/mobile/supply/S, atom/movable/AM, datum/economy/cargo_shuttle_manifest/manifest)
+	SIGNAL_HANDLER
+	return check_sell(S, AM) & COMSIG_CARGO_SELL_WRONG
+
+/datum/economy/simple_seller/proc/end_sell(obj/docking_port/mobile/supply/S, datum/economy/cargo_shuttle_manifest/manifest)
+	SIGNAL_HANDLER
+	return
+
+
+/datum/economy/simple_seller/crates
+	var/crates = 0
+	var/credits = 0
+
+/datum/economy/simple_seller/crates/begin_sell(obj/docking_port/mobile/supply/S)
+	SIGNAL_HANDLER
```

Comment:
```suggestion
```
"Should not sleep" doesn't reflect onto child functions. You only need to define it once on the function chain